lib/fog/core/provider.rb

```diff
@@ -21,11 +21,20 @@ def service(new_service, constant_string)
       Fog.services[new_service] ||= []
       Fog.services[new_service] |= [to_s.split("::").last.downcase.to_sym]
       @services_registry ||= {}
-      @services_registry[new_service] = [to_s, constant_string].join("::")
+      @services_registry[new_service] = service_klass(constant_string)
```

Comment:
I'm not sure how this is working now. As of my understanding it will always load services in the form of `Fog::<PROVIDER>::<Service>` and we have many providers doing `Fog::<Service>::<PROVIDER>`

Reply:
If I recall correctly, it was all `Fog::<PROVIDER>::<SERVICE>` initially, and we might have changed some along the way in the hopes that the other way was better? It would be best, I suspect, to pick one and stick to it though (which would probably be the old way). What do you think?

Reply:
I agree to sticking to one. So my idea is to deprecate the one that might give us problems (2 different providers define a class named `Fog::Compute::Utils`) and drop support in a major version. This is being done from the `service_klass` method.